Assets/Scripts/InteractionSystem/Interact.cs

```diff
@@ -0,0 +1,114 @@
+/******************************************************************
+*    Author: Marissa Moser
+*    Contributors: 
+*    Date Created: May 24, 2024
+*    Description: 
+*       Contains a function called when the player presses E. This function will:
+            Use a raycast to check if there is something in from of them
+            to interact with and if it is interactable. If so, call the
+            Interact(player) function on that game object
+
+*******************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class Interact : MonoBehaviour
+{
+    PlayerControls _playerControls;
+    private InputAction _interact;
+
+    [SerializeField] private GameObject _targetGameObj;
+    [SerializeField] private GameObject _interactable;
+    private bool _canInteract;
+
+    //raycast variables
+    private RaycastHit _colliderHit;
+    [SerializeField] private float _maxInteractDistance;
+    [SerializeField] LayerMask _layerToIgnore;
+
+    private void Awake()
+    {
+        _playerControls = new PlayerControls();
+        _playerControls.BasicControls.Enable();
+        _interact = _playerControls.FindAction("Interact");
+        _interact.started += InteractPressed;
+
+        StartDetectingInteractions();
+    }
+
+    /// <summary>
+    /// Called when Interact input is started. Calls Interact() on the detected
+    /// interactable game object
+    /// </summary>
+    private void InteractPressed(InputAction.CallbackContext ctx)
+    {
+        if(_interactable != null)
+        {
+            _interactable.GetComponent<IInteractable>().Interact(gameObject);
+        }
+          
+    }
+
+    /// <summary>
+    /// Starts the Detect Interactable coroutine
+    /// </summary>
+    public void StartDetectingInteractions()
+    {
+        _canInteract = true;
+        StartCoroutine(DetectInteractable());
+    }
+
+    /// <summary>
+    /// Ends the Detect Interactable coroutine
+    /// </summary>
+    public void StopDetectingInteractions()
+    {
+        _canInteract = false;
+    }
+
+    /// <summary>
+    /// A coroutine that detects if there is an interactable object in front of
+    /// the player using a raycast. This coroutine can be stopped with the public 
+    /// Start/StopDetectingInteraction function
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator DetectInteractable()
+    {
+        while(_canInteract)
+        {
+            //Casts Raycast in the center of the screen
+            Ray r = Camera.main.ScreenPointToRay(new Vector3(Screen.width / 2, Screen.height / 2, 0));
```

Comment:
Don't use camera.main

Reply:
Is there a reason for this? I understand why finding the camera is not great, but in this instance I just need to find the direction the camera is facing